sql-constants.md

```diff
@@ -0,0 +1,220 @@
+---
+title: SQL Constant values
+summary: SQL Constants represent a simple value that doesn't change.
+toc: false
+---
+
+SQL Constants represent a simple value that doesn't change.
+
+<div id="toc"></div>
+
+## Introduction
+
+There are five categories of constants in CockroachDB:
+
+- string literals: these define string values but their actual data type will
+  be inferred from context, for example `'hello'`;
+- numeric literals: these define numeric values but their actual data
+  type will be inferred from context, for example `-12.3`;
+- byte array literals: these define byte array values with data type
+  `BYTES`, for example `b'hello'`;
+- interpreted literals: these define arbitrary values with an explicit
+  type, for example `INTERVAL '3 days'`.
+- named constants: these have predefined values with a predefined
+  type, for example `TRUE` or `NULL`.
+
+## String literals
+
+CockroachDB supports multiple formats for string literals:
+
+- standard SQL string literals
```

Comment:
Let's make each of these link down to the appropriate section, i.e., `[standard SQL string literals](#standard-sql-string-literals)`